### PR TITLE
#11817 - Don't allow the saving of forms that have mandatory fields f…

### LINKED ISSUE #11817: Don't allow the saving of forms that have mandatory fields filled only with 'spaces' [3]
### Problem Description

Forms that contain mandatory fields can be saved if the user inputs spaces in the mandatory fields.

### Proposed Change

Do not allow the saving of forms when only spaces have been inputted in the mandatory firlds.

### Added Value/Benefit

As long as it's possible to only enter spaces, users can circumvent text fields being mandatory which results in missing information. This change will ensure that this is no longer possible.

### Acceptance Criteria

- [ ] The `@NotBlank` annotation is used instead of `@NotNull` or `@NotEmpty` for all String fields in DTOs that are currently using the `@NotNull` or `@NotEmpty` annotation:
  - DocumentDto.name
  - DocumentDto.mimeType
  - DocumentDto.relatedEntityUuid
  - EventDto.name
  - EventGroupDto.name
  - PersonDto.firstName
  - PersonDto.lastName
  - ExternalVisitDto.personUuid
  - SormasToSormasOptionsDto.comment
  - SormasToSormasOriginInfoDto.comment
- [ ] Forms that have text fields marked as mandatory can no longer be saved when only putting in blank spaces
- [ ] Instead, an error message appears that the field is a required field
- [ ] Additionally, `isBlank` is used instead of `isEmpty` when validating the dearchive reason field in `ArchivingController.dearchiveEntity`, and it is no longer possible to specify a dearchive reason consisting only of blank spaces
- [ ] Users with pseudonymized access to data (e.g. a person) can still save the respective form without getting an error (implementing the changes above will otherwise likely result in a validation error that, e.g. in a person form, first and last name are missing) AND the respective fields are unchanged afterwards (e.g. the person first name is still the same as before)

### Implementation Details

In order to fulfill the last acceptance criteria, there are two possible options that have to be tried out. In order of preference, those are:

1. Create a new interceptor that checks whether a PseudonymizableDto field is annotated with `@NotBlank` AND the Dto is `pseudonymized == true`, and if so, sets the value of this field to "CONFIDENTIAL".
2. Change the pseudonymizer to not replace String fields annotated with `@NotBlank` with an empty String, but use the String "CONFIDENTIAL" instead.

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
…illed only with 'spaces'

Fixes #11817

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12059

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java [hunk 1]
```diff
@@ -185,8 +188,8 @@ private void assertNotPseudonymized(EventParticipantDto eventParticipant) {
 
 	private void assertPseudonymized(EventParticipantDto eventParticipant) {
 		assertThat(eventParticipant.getInvolvementDescription(), isEmptyString());
-		assertThat(eventParticipant.getPerson().getFirstName(), isEmptyString());
-		assertThat(eventParticipant.getPerson().getLastName(), isEmptyString());
+		assertEquals(I18nProperties.getCaption(Captions.inaccessibleValue), eventParticipant.getPerson().getFirstName());
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java:191]: please use assertThat to don't break the logic

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java [hunk 2]
```diff
@@ -185,8 +188,8 @@ private void assertNotPseudonymized(EventParticipantDto eventParticipant) {
...
+		assertEquals(I18nProperties.getCaption(Captions.inaccessibleValue), eventParticipant.getPerson().getLastName());
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java:192]: same

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/NotBlankTextValidator.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+public class NotBlankTextValidator extends AbstractValidator<String> {
+
+	public NotBlankTextValidator(String errorMessage) {
+		super(errorMessage);
+	}
+
+	@Override
+	protected boolean isValidValue(String s) {
+
+		return !s.isBlank();
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/NotBlankTextValidator.java:14]: Please use `StringUtils.isNotBlank`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/page/title/TitleLayoutHelper.java [hunk 4]
```diff
@@ -12,7 +12,11 @@ private TitleLayoutHelper() {
 
 	public static StringBuilder buildPersonString(PersonDto person) {
 		StringBuilder personString = new StringBuilder();
-		String personFullName = PersonDto.buildCaption(person.getFirstName(), person.getLastName());
+		String personFullName = "";
+		if (!person.isPseudonymized()) {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/page/title/TitleLayoutHelper.java:16]: if the person is pseudonymized then the fillname should be "Confidential" I would say

File: sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java [hunk 5]
```diff
@@ -521,8 +522,8 @@ private void assertNotPseudonymized(PersonDto person) {
 	private void assertPseudonymised(PersonDto person) {
 
 		assertThat(person.isPseudonymized(), is(true));
-		assertThat(person.getFirstName(), isEmptyString());
-		assertThat(person.getLastName(), isEmptyString());
+		assertEquals(I18nProperties.getCaption(Captions.inaccessibleValue), person.getFirstName());
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java:525]: same as above

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java:191]: updated

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/NotBlankTextValidator.java (hunk 3 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/NotBlankTextValidator.java:14]: updated

File: sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjbPseudonymizationTest.java:192]: updated

File: sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java (hunk 5 above)
Comment (sergiupacurariu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/person/PersonFacadeEjbPseudonymizationTest.java:525]: updated

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/page/title/TitleLayoutHelper.java (hunk 4 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/page/title/TitleLayoutHelper.java:16]: updated

**Review by leventegal-she — APPROVED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12059

**Pull request merged**